### ISSUE TITLE
Thin (10nm) Quantum Well (QW), core appears triangular

### ISSUE THREAD

**Issue by neonot** [jychstar/optical_waveguide#1]
For a thin (say 10nm) QW, does one need to invoke anything special in the code? The QW appears triangular if set to 10nm. I am trying to check the code for a structure similar to 10.1109/ICIPRM.1994.328215  for a 10nm QW.

**Comment by jychstar** [comment]
I am not sure what specific inputs you use.  For a thin QW which is about 2 orders of magnitude smaller than the wavelength, I expect the majority of optical wave leak out of core layer.  You have to use a waveguide thick enough to confine the wave and get a converged solution.

**Comment by neonot** [comment]
my design is as follows, QW 10nm, Waveguide 120nm each on either side of the QW, cladding 2um each on either side of the waveguide. Effective index guess = 3.28. QW index =3.6., Waveguide index =3.27, Cladding index = 3.191. I get a triangular QW shape in the matlab plot generated by your code. Is this expected?

**Comment by jychstar** [comment]
I use your input parameters and get this:
![untitled](https://cloud.githubusercontent.com/assets/20079088/19244094/56bacf7c-8ee0-11e6-84a0-de53235439e5.png)
Looks good to me. Is this what you mean by triangular shape?

**Comment by neonot** [comment]
Thanks your AnExample2.m works fine. However, the code below generates the figure pasted below.

``
%% input waveguide parameter
clc
clear all;
close all;
tic % time program run time

TE=1; % TE=1 for TE mode, TE=0 for TM mode
lambda=0.63; % um, emitting wavelength
k0=2*pi/lambda;  % wave-vector, in 1/um

Nr=1001;  % number of points to get the real part of effective index
Ni=1001;  % number of points to get the imaginary part of effective index
xr=linspace(3.2,3.4,Nr); % generate Nr points in an estimated range of index
xi=-linspace(-1,1,Ni)/k0/2*1e-4;   % range of loss in cm^-1, loss= 2*k0*imag(complex index)

% assign refractive index to specific material or region
CoreQWIndx=3.6;
WaveguideIndx = 3.27;
CladdingIndx = 3.191;
Cladding2Indx = 3.191;


% assign refractive index to each layer
r1=Cladding2Indx ;
r2=CladdingIndx;
r3=WaveguideIndx;
r4=CoreQWIndx;
r5=WaveguideIndx;
r6=CladdingIndx;
r7=Cladding2Indx ;

% assign thickness in um to each layer
t1=1;
t2=1;
t3=0.12;
t4=0.01;  % Set to the QW
t5=t3;
t6=t2;
t7=t1;

layer_grid = 0.01;

optical_field_file='optical_field.txt';  % assign file name for output
far_field_file='far_field.txt';

% build waveguide structure 
thick=[t1,t2,t3,t4,t5,t6,t7];% thickness of each layer, in um
ref=[r1,r2,r3,r4,r5,r6,r7];
active_layer_sequence = 4;  % indicate which layer is active region, from top

......
``
![capture](https://cloud.githubusercontent.com/assets/16015258/19266192/48a1e54a-8fa0-11e6-9a4f-2d20ecd03bdc.JPG)


My question on the triangular shape of the QW is related to the fact that in the figure window, the QW appears as triangular (in red) and not as a rectangle?

**Comment by jychstar** [comment]
Note that layer_grid = 0.01, so you only have one point plot for the QW then you will have triangular.

**Comment by neonot** [comment]
thanks -makes sense.
It would be useful to understand why the code I have pasted above leads to a extra notch between 3 and 3.5 um. Also, what is your definition of the "Top"?
Regards,

**Comment by jychstar** [comment]
Top is actually the first layer of your input.  I usually imagine it's an epi-side up device.
The notch is probably the accuracy is not good enough. You may try narrow down the estimated range, etc.

**Comment by neonot** [comment]
thanks. In your far-field function you have
field.*exp(-1i*k0*depth*sintheta(k))

should it not be 
field.*exp(-1i*k0*depth*sintheta*k);

The integral is over _k_ and not sinetheta

?

**Comment by jychstar** [comment]
k represents the angle. If you integrate over k, you won't get angle-dependent far-field.
 
The codes you referred: 
for k=1:1:length(theta)
    fun=field.*exp(-1i*k0*depth*sintheta(k));  %  the integrand **fun** is evaluated at **angle k**
    Efar(k)=trapz(depth,fun);  % integral is over **depth**
end
